src/test/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesTaskLauncherIntegrationIT.java

```diff
@@ -16,235 +16,122 @@
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
 
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.cloud.deployer.resource.docker.DockerResource;
 import org.springframework.cloud.deployer.spi.core.AppDefinition;
 import org.springframework.cloud.deployer.spi.core.AppDeploymentRequest;
 import org.springframework.cloud.deployer.spi.task.LaunchState;
-import org.springframework.cloud.deployer.spi.task.TaskLauncher;
-import org.springframework.cloud.deployer.spi.test.AbstractTaskLauncherIntegrationJUnit5Tests;
-import org.springframework.cloud.deployer.spi.test.Timeout;
 import org.springframework.core.io.Resource;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.entry;
-import static org.awaitility.Awaitility.await;
 
 /**
  * Integration tests for {@link KubernetesTaskLauncher}.
  *
  * @author Thomas Risberg
  * @author Chris Schaefer
+ * @author Chris Bono
  */
 @SpringBootTest(classes = {KubernetesAutoConfiguration.class}, properties = {
 		"spring.cloud.deployer.kubernetes.namespace=default"
 })
-public class KubernetesTaskLauncherIntegrationIT extends AbstractTaskLauncherIntegrationJUnit5Tests {
-
-	@Autowired
-	private TaskLauncher taskLauncher;
-
-	@Autowired
-	private KubernetesClient kubernetesClient;
-
-	@Override
-	protected TaskLauncher provideTaskLauncher() {
-		return taskLauncher;
-	}
+public class KubernetesTaskLauncherIntegrationIT extends AbstractKubernetesTaskLauncherIntegrationTests {
 
 	@Test
-	@Override
-	@Disabled("Currently reported as failed instead of cancelled")
-	public void testSimpleCancel() throws InterruptedException {
-		super.testSimpleCancel();
-	}
-
-	@Override
-	protected String randomName() {
-		return "task-" + UUID.randomUUID().toString().substring(0, 18);
-	}
-
-	@Override
-	protected Resource testApplication() {
-		return new DockerResource("springcloud/spring-cloud-deployer-spi-test-app:latest");
-	}
-
-	@Override
-	protected Timeout deploymentTimeout() {
-		return new Timeout(20, 5000);
+	void taskLaunchedWithJobPodAnnotations(TestInfo testInfo) {
+		logTestInfo(testInfo);
+		launchTaskPodAndValidateCreatedPodWithCleanup(
+				Collections.singletonMap("spring.cloud.deployer.kubernetes.jobAnnotations", "key1:val1,key2:val2,key3:val31:val32"),
+				(pod) -> {
+					assertThat(pod.getSpec().getContainers()).isNotEmpty()
+							.element(0).extracting(Container::getPorts).asList().isEmpty();
+					assertThat(pod.getMetadata().getAnnotations()).isNotEmpty()
+							.contains(entry("key1", "val1"), entry("key2", "val2"), entry("key3", "val31:val32"));
+				});
 	}
 
 	@Test
-	public void testJobPodAnnotation() {
-		log.info("Testing {}...", "JobPodAnnotation");
-
-		KubernetesTaskLauncher kubernetesTaskLauncher = new KubernetesTaskLauncher(new KubernetesDeployerProperties(),
```

Comment:
Also notice that we were constructing this local KTL in order to pass in the KDP that we modified in the job-based IT but do not do anything to in the pod-based IT. This is really not needed since this is a `@SpringBootTest` that is auto-configured w/ `KubernetesAutoConfiguration` which results in a KTL that is plenty sufficient to use. In the job-based IT, a `@SpringBootTest( properties = "` is what varies the KTL between job/pod based tasks. 

This also allows us to get rid of some extra code.